### PR TITLE
bpo-41675: Modernize siginterrupt calls

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue41675

### CODE REVIEW

File: Modules/signalmodule.c [hunk 1]
```diff
@@ -664,7 +664,19 @@ signal_siginterrupt_impl(PyObject *module, int signalnum, int flag)
...
+#ifdef HAVE_SIGACTION
+    struct sigaction act;
+    (void) sigaction(signalnum, NULL, &act);
+    if (flag) {
+        act.sa_flags &= ~SA_RESTART;
+    }
+    else {
+        act.sa_flags |= SA_RESTART;
+    }
+    if (sigaction(signalnum, &act, NULL)<0) {
```
Comment (vstinner) [inline Modules/signalmodule.c]: nitpick:

```suggestion
    if (sigaction(signalnum, &act, NULL) < 0) {
```

and also below: `if (siginterrupt(signalnum, flag) < 0) {`.

File: Misc/NEWS.d/next/Core and Builtins/2020-08-31-14-53-17.bpo-41675.VSoqWU.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+The implementation of :func:`signal.siginterrupt` now uses
+:c:func:`sigaction` instead of the deprecated :c:func:`siginterrupt`. Patch
```
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2020-08-31-14-53-17.bpo-41675.VSoqWU.rst]: You may mention that it's only used when sigaction() is available.

File: Modules/signalmodule.c (hunk 1 above)
Comment (pablogsal) [reply Modules/signalmodule.c]: Done

File: Misc/NEWS.d/next/Core and Builtins/2020-08-31-14-53-17.bpo-41675.VSoqWU.rst (hunk 2 above)
Comment (pablogsal) [reply Misc/NEWS.d/next/Core and Builtins/2020-08-31-14-53-17.bpo-41675.VSoqWU.rst]: Done

**Review by vstinner — APPROVED** [review]
LGTM.

The glibc deprecated siginterrupt() function has exactly the same code!
https://sourceware.org/git/?p=glibc.git;a=blob;f=sysdeps/posix/sigintr.c;h=dd8815a3bcfd1943b5a45933bfbe008fa3d6bee5;hb=HEAD

**Pull request merged**